doc/BuildingWithCMake.md

```diff
@@ -54,6 +54,104 @@ CMake can generate build files for a number of backends. You can select the gene
 
+## Configuring the build on Windows
+
+There are three widely used C/C++ toolchains for Windows: *MSYS2/MinGW*, *Microsoft Visual Studio* and *LLVM/Clang*.
+Let's consider how to use each toolchain to configure and build `OMR` with `CMake`.
+
+### Common prerequisite
+
+*Perl* must be installed otherwise `CMake` won't be available to configure the project:
+
+```bash
+-- Could NOT find Perl (missing: PERL_EXECUTABLE)
+CMake Error at cmake/modules/OmrCompilerSupport.cmake:42 (message):
+  Perl not found
+```
+
+[Strawberry Perl](http://strawberryperl.com/) is absolutelly suitable for `CMake` on Windows, please do not forget
+to add the path to `perl.exe` (for example `C:\StrawberryPerl\perl\bin`) to the `PATH` environment variable.
+
+### Configuring the build for using MSYS2/MinGW
+
+The code of `OMR` uses the Microsoft extension to `C` for supporting *Windows Structured Exception Handling
+([SEH](https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms680657(v=vs.85).aspx))*. *MinGW* haven't supported it
+for a long time. Some workarounds exist but none of them is suitable to use in production-quality software.
+
+### Configuring the build for using Visual Studio Build Tools
+
+*Visual Studio Build Tools* is a bundle with a standalone compiler, libraries and scripts to build C++ libraries and
+applications targeting Windows desktop. The [build tools](http://landinghub.visualstudio.com/visual-cpp-build-tools)
+are available for download on Microsoft's website.
+
+The following components of the build tools are recommended to be installed:
+
+ * Windows 10 SDK (>= 10.0.16299.0) for Desktop C++
+ * Visual C++ tools for CMake
+
+**Note:** 
+To use the `NMake` or `Ninja` generator with *Visual C++*, `cmake` must be run from a shell that can use 
+the compiler `cl` from the command line. To enable the shell, please run
+`C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvarsall.bat [platform]` where `platform`
+can be one of the following options: `x86_amd64`, `x86_arm uwp`, `x86_arm onecore`, `x64 8.1` or `x64 store 8.1`.
+
+**Note:**
+If you wish to build the `DDR` component, `CMake` should be able to find the path to the `DIA SDK`. There are two ways to specify
+the path:
+
+ * Create the `VSSDK140Install` environment variable and set its value to the path to the *Visual Studio Build Tools* installation,
+for example, `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools`.
+
+ * Create the `DIASDK` environment variable set its value to the path to the `DIA SDK` folder itself, for example,
+`C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\DIA SDK`.
+
+Then, to configure the build, let's do this:
+
+```bash
+# always prefer building outside the source tree
+mkdir -p build && cd build
+
+# configure the build using Ninja
+cmake -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_COMPILER=cl .. -G"Ninja"
+
+# ... or using NMake
+cmake -DCMAKE_CXX_COMPILER=cl -DCMAKE_C_COMPILER=cl .. -G"NMake Makefiles"
+
```

Comment:
On Windows you can also have CMake generation a Visual Studio Solution File and use MSBuild to build it.  This is how the automated AppVeyor builds run at the project.

You can do this via the following command line as an example.  You can find all of the msbuild generators here (https://cmake.org/cmake/help/v3.6/manual/cmake-generators.7.html#id10)
cmake -G"Visual Studio 11 2012 Win64"
cmake --build (or msbuild omr.sln)

Reply:
@charliegracie Do you wish me to add this information to the document or you wish to do it itself after the merge? 

One comment only: `cmake -G"Visual Studio 15 2017 Win64"` (cmake 3.11.0-rc3), it works fine.

If any new comments are expected, should I add the `WIP` to the PR's name?

Reply:
I can add it afterwards. I am just waiting for a review from one of the developers more familiar with CMake.